include/fmt/core.h

```diff
@@ -105,9 +105,11 @@
 #  define FMT_CONSTEXPR
 #endif
 
-#if ((FMT_CPLUSPLUS >= 202002L) &&                            \
-     (!defined(_GLIBCXX_RELEASE) || _GLIBCXX_RELEASE > 9)) || \
-    (FMT_CPLUSPLUS >= 201709L && FMT_GCC_VERSION >= 1002)
+#if (FMT_CPLUSPLUS >= 202002L ||                                \
+     (FMT_CPLUSPLUS >= 201709L && FMT_GCC_VERSION >= 1002)) &&  \
+    ((!defined(_GLIBCXX_RELEASE) || _GLIBCXX_RELEASE >= 10) &&  \
+     (!defined(_LIBCPP_VERSION) || _LIBCPP_VERSION >= 10000) && \
+     (!FMT_MSC_VERSION || FMT_MSC_VERSION >= 1928))
```

Comment:
What is the idea behind this change? Shall we enable `FMT_CONSTEXPR20` only if `is_constant_evaluated` is available instead? Something like

```c++
#if defined(FMT_HAS_IS_CONSTANT_EVALUATED) &&                 \
    ((FMT_CPLUSPLUS >= 202002L) &&                            \
     (!defined(_GLIBCXX_RELEASE) || _GLIBCXX_RELEASE > 9)) || \
    (FMT_CPLUSPLUS >= 201709L && FMT_GCC_VERSION >= 1002)
#  define FMT_CONSTEXPR20 constexpr
```

Reply:
Main idea: Enable C++20 constexpr only if stdlib support C++20.
The condition ``(!defined(_GLIBCXX_RELEASE) || _GLIBCXX_RELEASE > 9)`` partially check stdlib C++20 features. I extended it for libcxx and MSVC.

``FMT_CONSTEXPR20`` enabled if compiler support C++20 AND the version of stdlib is modern to support C++20.

Reply:
The compiler does not define _LIBCPP_VERSION in Android's case. 
This will result in FMT_CONSTEXPR20 being incorrectly set to constexpr.

Reply:
@hchataing

Try to find the ``__config`` file in the source code of your standard library and look at the value of the ``_LIBCPP_VERSION`` macro.

Or in repository, like this:
https://android.googlesource.com/platform/external/libcxx/+/refs/tags/android-cts-10.0_r15/include/__config#36

Reply:
:+1:  I see `#  define _LIBCPP_VERSION 170000`

Reply:
@hchataing
Interesting!
libcxx version 170000 fully supports C++20 and you shouldn't have issue #3745.

Can you compile and run the program:
```c++
#include <iostream>

int main()
{
    std::cerr << _LIBCPP_VERSION << std::endl;
    return 0;
}
```

in issue #3745 build environment?

Reply:
I think we should just go with enabling C++20 constexpr only if `consteval` is also available.

Reply:
Done.

Reply:
Sorry, I meant `is_constant_evaluated` because that's what causing the OP's problem.

Reply:
@vitaut 
Done.